### ISSUE TITLE
Bug: Cancellation regression

### ISSUE THREAD

**Issue by LeoSht** [fsharplang-ru/pulsar-client-dotnet#200]
Hi,
The following test passes when compiled against the latest "develop" branch source code, but fails if compiled against the NuGet starting with v.2.9.2:
```csharp
[Fact]
public async Task ConsumerReceiveWithCancellation_WithoutAnyMessages()
{
    // Arrange
    Log.Debug("Started Cancellation without receiving a single message works fine");
    var client = await new Pulsar.Client.Api.PulsarClientBuilder()
                            .ServiceUrl("pulsar://127.0.0.1:6650")
                            .BuildAsync();

    var topicName = "public/default/topic-" + Guid.NewGuid().ToString("N");
    var cts = new CancellationTokenSource();
    var consumer = await client.NewConsumer()
            .Topic(topicName)
            .SubscriptionName("test-subscription")
            .SubscribeAsync();

    // Act
    var consumerTask = consumer.ReceiveAsync(cts.Token);
    cts.CancelAfter(100);

    // Assert
    try
    {
        await consumerTask;
        Assert.True(false, "Unexpected task success.");
    }
    catch (Exception ex)
    {
        Log.Debug(ex, "Exception:");
    }

    Assert.Equal(TaskStatus.Canceled, consumerTask.Status);
    Log.Debug("Finished Cancellation without receiving a single message works fine.");
}
```

**Comment by Lanayx** [comment]
Thank you for creating an issue. I was able to reproduce the issue and found out that the issue is connected with Release build mode and new F# 6 tasks builder, however I hasn't been able to make a simple repro to create issue in fsharp repository. Will keep you updated on the issue as I move forward with it.

**Comment by Lanayx** [comment]
@LeoSht This should be fixed in 2.9.4, can you please check

**Comment by LeoSht** [comment]
@Lanayx Thank you for the quick turnaround. I confirmed that the regression is fixed in 2.9.4.
